### ISSUE TITLE
Undoing Accepting a Mission

### ISSUE THREAD

**Issue by benurban** [endless-sky/endless-sky#10343]
### Problem Description

I don't know how often this functionality might actually be used, but there seems to be a general feeling in this game that almost any action is undoable if undone immediately.

In this case, I'm referring to accepting missions. If you take on a (regular) mission, and abort it before you leave, of course there is no penalty. (In fact there is no penalty even if you abort it later, but #9670, for instance, would introduce such a penalty.) But the mission can no longer be accepted again.

### Related Issue Links

I found no other issues that discussed this.

### Desired Solution

I feel that a mission should return to the list if it is aborted before departing.

### Alternative Approaches

Not aborting the mission at all comes to mind, but one could change one's mind...

**Comment by mOctave** [comment]
I think whether this makes sense or not depends a lot on the mission. If its a routine job, or another minor mission that just needs to get done eventually, then it makes sense to be able to have it offered again (which, of course, is often done with repeating jobs). But if it's a major campaign mission, then you've just accepted a job that's going to have galaxy-wide consequences for all of humanity, walked out of the room, turned around and clearly cancelled your participation—I can't see a good lore or gameplay reason why you would then be _allowed_ to change your mind: you've already proven that you're unreliable.
I think the conversation aspect of the campaign missions is designed to give some sense of permanency to your actions. After all, you can't just decline a mission, then realize you needed it and click on the spaceport again to go back and accept it. So why should you be able to do the same thing after you had already accepted it?

**Comment by benurban** [comment]
I absolutely agree. I was only referring to missions that came from the job board in the first place.

For major missions, I will offer one request: Make it possible to undo a single step in the conversation, or add a blank line between the options (I think this wou\d be a better choice). I, for one, am often worried about missing the choice I wanted to select on my tiny touchscreen because it's smaller than my finger.

**Comment by benurban** [comment]
> In fact there is no penalty even if you abort it later

I just learned today that some routine missions have (or at least claim to have) a penalty for aborting. I accidentally accepted one of the pirate missions, and immediately aborted it. The game said the pirates wouldn't be happy. I think this should also be fixed, for the same reason.

**Comment by Saugia** [comment]
What would the difference between having an unaccepted mechanic for and reloading your last save file? It seems like that already fulfills what you wish to see here. I know it sounds a bit silly, but part of the reason you can reload the save file is so you can take a step back if you need to, up to three landings ago, or last spaceport. I'm not sure why we would need a specific mechanic to do this otherwise when we already have the means to do it, especially if it's only one click less.

**Comment by benurban** [comment]
As I am new to this game and its philosophy, I will answer that question by asking my own: What is the reasoning behind allowing the player to undo purchases and sales (at leat of ships and outfits) without penalty, if loading the last save is a solution?

For what it's worth, I did wind up loading a save for that pirate mission. I don't like doing that any more than I have to, though. It feels very cheaty to me, and it's very easy to get into that habit.

**Comment by Saugia** [comment]
I suppose that is a different manner. You're outfitting your ship and purchasing or selling a given outfit in the outfitter - you can buy something for a price, and if you no longer decide you wanted it, sell it back to the shop (before taking off). If you're selling, you could sell something, but decide to take it back. It is indeed partly convenience, though I feel outfitting your ship is not the same as accepting a job to perform for someone that leads elsewhere on the map and involves more individuals than just you or the person you are talking to.

One could change one's mind on an outfit purchase for a ship, but ship outfitting is more involved in the game than minor jobs that appear all the time - selling an alien outfit but changing your mind to want it back before taking off on a planet that would never sell such an item is not the same as changing your mind on a minor job that always has a chance to appear again on the planet (or any of those near it).

If you agree to do work for a faction, and suddenly they find out you declined it right after meeting them, they would get understandably upset at that notification and probably not want to offer it to you again, and would be busy as jobs are pretty common to come across.

So my answer would be, jobs are not unique, but outfitting your ship with outfits, especially ones that you cannot obtain otherwise or elsewhere, is.

**Comment by Saugia** [comment]
I think the better answer, or alternative to discuss, is to decide whether or not missions should hurt you on abort as opposed to fail, or only hurt you if you abort them after taking off.

**Comment by benurban** [comment]
I do think that is absolutely worth discussing, now that I know that aborting some routine missions can hurt, as things currently stand.

**Comment by benurban** [comment]
As far as the original issue, I don't think making it undoable makes sense from an in-game perspective; I'm thinking about it from a usability perspective.

On the same note, though: In-game, I don't think Joe Outfitter is likely to allow you to buy back your fancy alien hardware without penalty, either.

**Comment by Saugia** [comment]
Yeah; that's why I'm not suggesting we shouldn't do it, more-so asking how much benefit does it bring to add more code in when it's already doable. My comments are mostly on comparing the ability to buyback alien outfits and such is far more important than unaccepting a job you can find the next day.

**Comment by benurban** [comment]
Yeah, like I said, I don't know how often someone will want to accept the same job again, or care if they can't. Such a capability definitely falls more under "nice to have."

As for the other, in my experience, it is easy to accidentally hit accept one too many times if you are, for instance, accepting all jobs from one system.

**Comment by mOctave** [comment]
> add a blank line between the options

Actually, this seems like a great idea, maybe not as something that's always enabled, but as an accessibility setting.

**Comment by Saugia** [comment]
Would implementing https://github.com/endless-sky/endless-sky/pull/9824 be an alternative here?

**Comment by benurban** [comment]
I don't see how it would hurt, but I'm not sure what it would solve with regards to this issue. Unless you meant for it to give a warning before aborting a mission when there's a penalty? I think if the penalty is to be kept even for as immediate abort, the warning would be needed when accepting the mission.

**Comment by Saugia** [comment]
Having an abort warning for rep hits for aborting a mission would resolve the conern that you can't unabort a mission that hurts your rep, so I think that's the better alternative here.

As for aborting a job and it not re-appearing in the job board, that's intended gameplay. You're offered a job, walk, away, and then change your mind on it; you're probably not going to get the same job back if you already have denied it, and they wouldn't be interested in having it available to the individual that already declined it. Outfitters are different; we could say that even though you purchased the outfit, you could be testing it before you use it, or leave, and the price could be insurance nothing happens. Or, perhaps purchases aren't finalized until you leave the planet.

In either case, jobs not offering again after being aborted is intentional.

**Comment by benurban** [comment]
What I meant was that (at least in the case of that pirate mission) there is a cost (or risk) one way or the other once the mission is accepted, so that would be the appropriate time for a warning. I don't object to adding a warning when trying to abort, as well. But it would be worth requiring confirmation when accepting a mission that is illegal, unless the player already has a bad rep with local law enforcement.

I haven't yet encountered a mission that carries no such risk, but penalizes aborting. If such a mission exists, a warning for aborting such a mission would suffice.

**Comment by derei** [comment]
Suggested approach:
- As long as the player is still in the Job Board, cancelling a mission immediately after accepting it, should not generally bring consequences. To give  it a twist, let's say that some super urgent and volatile missions would disappear upon cancelling (a different captain snatched it when you canceled). This would help with accidental mission selection.
- HOWEVER, aborting a mission AFTER leaving the Job Board, but BEFORE leaving the Spaceport (eg. player returns to Job Board, should have lesser consequences compared to Aborting the mission after leaving the planet (when is technically considered you're on route).

Thoughts?

**Comment by benurban** [comment]
That sounds like it would work. It's a bit more complicated, but it might
be worth doing.

**Comment by Just-Existing** [comment]
>* HOWEVER, aborting a mission AFTER leaving the Job Board, but BEFORE leaving the Spaceport, (eg. player returns to Job Board, should have lesser consequences compared to Aborting the mission after leaving the planet (when is technically considered you're on route).

That seems *way* too complicated. I don't think the game has any mechanism to detect which part of the UI the player is in. However, allowing any mission to be aborted without consequences if it is aborted before the player leaves the planet seems worthwhile and significantly more doable.

**Comment by QuantumTux** [comment]
See also [Issue #9670 my proposal for a Commercial Reputation](https://github.com/endless-sky/endless-sky/issues/9670)
